lib/azure/armrest/storage_account_service.rb

```diff
@@ -146,7 +197,11 @@ def list_private_images(group = configuration.resource_group)
 
         list(group).each do |lstorage_account|
           threads << Thread.new(lstorage_account) do |storage_account|
-            key = list_account_keys(storage_account.name, group).fetch('key1')
+            if Time.parse(api_version).utc >= Time.parse('2016-01-01').utc
+              key = list_account_key_objects(storage_account.name, group).first.key
+            else
+              key = list_account_keys(storage_account.name, group).fetch('key1')
```

Comment:
I trusted you have tested this. Just curious, `key` is either a `Hash` or a `Model`, can it properly work when it later is used by `storage_account`?